### ISSUE TITLE
Mismatch in log_probs dim and vocab dim?

### ISSUE THREAD

**Issue by sumeet-tiwari-samespace** [NVIDIA/NeMo#1188]
Post transcription I checked log_probs size and got this
`result['log_probs'].size()`
Got
`torch.Size([1, 7528, 29])`

Using Google Collab

`print(len(quartznet.decoder.vocabulary))`
Got
28

Why is there a mismatch of 28 and 29?

**Comment by khursani8** [comment]
"Why is there a mismatch of 28 and 29?"
I think because CTC blank token not in vocabulary

**Comment by okuchaiev** [comment]
yes, CTC black token is a special one and it is part of the vocabulary. There are, however, a dim for it in the output logits.
